From 619000a3c5a57e5314ee3a0fbb8aae0dc485e8aa Mon Sep 17 00:00:00 2001
From: Simon Kelley <simon@thekelleys.org.uk>
Date: Wed, 29 Apr 2020 00:09:58 +0100
Subject: [PATCH 10/12] Suppress logging of listen addresses during startup.

The initial call to enumerate_interfaces() happens before the
logging subsystem in initialised and the startup banner logged.
It's not intended that syslog be written at this point.

Signed-off-by: Kevin Darbyshire-Bryant <ldir@darbyshire-bryant.me.uk>
---
 src/network.c | 27 +++++++++++++++++----------
 1 file changed, 17 insertions(+), 10 deletions(-)

diff --git a/src/network.c b/src/network.c
index 82201e3..c7d002b 100644
--- a/src/network.c
+++ b/src/network.c
@@ -717,7 +717,7 @@ int enumerate_interfaces(int reset)
 	  else if (release_listener(l))
 	    {
 	      *up = tmp;
-		freed = 1;
+	      freed = 1;
 	    }
 	}
 
@@ -1034,15 +1034,20 @@ void create_bound_listeners(int dienow)
 	  }
 	else if ((new = create_listeners(&iface->addr, iface->tftp_ok, dienow)))
 	  {
-	    int port;
-
 	    new->iface = iface;
 	    new->next = daemon->listeners;
 	    daemon->listeners = new;
 	    iface->done = 1;
-	    port = prettyprint_addr(&iface->addr, daemon->addrbuff);
-	    my_syslog(LOG_DEBUG, _("listening on %s(#%d): %s port %d"),
-		      iface->name, iface->index, daemon->addrbuff, port);
+
+	    /* Don't log the initial set of listen addresses created
+               at startup, since this is happening before the logging
+               system is initialised and the sign-on printed. */
+            if (!dienow)
+              {
+		int port = prettyprint_addr(&iface->addr, daemon->addrbuff);
+		my_syslog(LOG_DEBUG, _("listening on %s(#%d): %s port %d"),
+			  iface->name, iface->index, daemon->addrbuff, port);
+	      }
 	  }
       }
 
@@ -1061,12 +1066,14 @@ void create_bound_listeners(int dienow)
     if (!if_tmp->used && 
 	(new = create_listeners(&if_tmp->addr, !!option_bool(OPT_TFTP), dienow)))
       {
-	int port;
-
 	new->next = daemon->listeners;
 	daemon->listeners = new;
-	port = prettyprint_addr(&if_tmp->addr, daemon->addrbuff);
-	my_syslog(LOG_DEBUG, _("listening on %s port %d"), daemon->addrbuff, port);
+
+	if (!dienow)
+	  {
+	    int port = prettyprint_addr(&if_tmp->addr, daemon->addrbuff);
+	    my_syslog(LOG_DEBUG, _("listening on %s port %d"), daemon->addrbuff, port);
+	  }
       }
 }
 
-- 
2.24.2 (Apple Git-127)

